subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/gradleplugin/build.gradle.template

```diff
@@ -0,0 +1,59 @@
+/*
+ * This build file was auto generated by running 'gradle init --type gradle-plugin'
+ * by '${genUser.groovyComment}' at '${genDate.groovyComment}' with ${genGradleVersion.groovyComment}
+ *
+ * @author ${genUser.groovyComment}, @date ${genDate.groovyComment}
+ */
+
+// Apply the groovy plugin to also add support for Groovy (needed for Spock)
+apply plugin: 'groovy'
+// Apply the development plugin to make it easier to develop Gradle plugins
+apply plugin: 'java-gradle-plugin'
+
+// In this section you declare where to find the dependencies of your project
+repositories {
+    // Use 'jcenter' for resolving your dependencies.
+    // You can declare any Maven/Ivy/file repository here.
+    jcenter()
+}
+
+// Functional tests using Gradle TestKit.
+// Read more about this in the user guide ${ref_userguide_testkit.groovyComment}
+sourceSets {
```

Comment:
IMHO setting up the functional test source set + conventions should be provided by the Java Gradle plugin. We already declare the TestKit dependency automatically. Why not also setting up appropriate source set conventions?